### PR TITLE
Improved docs on debugging and monitoring the operator

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

File: content/docs/1.17/operator.md [hunk 1]
```diff
@@ -1013,13 +1013,48 @@ metadata:
...
+controller_runtime_reconcile_total{controller="jaeger-controller",result="error"}
+controller_runtime_reconcile_total{controller="jaeger-controller",result="success"}
+
+# Total number of retries (cumulative)
+workqueue_retries_total{name="jaeger-controller"}
+
+# Current number of reconciliation loops in progress (gauge)
+workqueue_depth{name="jaeger-controller"}
+
+# How many seconds of work has done that is in progress and hasn’t been observed by work_duration. (gauge)
```
Comment (objectiser) [inline content/docs/1.17/operator.md]: Is it possible to express this is a clearer way? Feel free to explain at length and I can try to provide a more concise definition.

File: content/docs/1.17/operator.md [hunk 2]
```diff
@@ -1013,13 +1013,48 @@ metadata:
...
+workqueue_unfinished_work_seconds{name="jaeger-controller"} 0
+
+# How long in seconds processing an item from workqueue takes. (bucket)
```
Comment (objectiser) [inline content/docs/1.17/operator.md]: "How long (in seconds) it takes to process an item from the workqueue. (bucket)"

File: content/docs/1.17/operator.md [hunk 3]
```diff
@@ -1013,13 +1013,48 @@ metadata:
...
+```
+
+A low number of reconciliation errors is normal, as there might be several processes changing resources at the same time for different reasons. Whenever there's a failure, the operator will attempt to reconcile again, increasing the `workqueue_retries_total` metric. However, if the rate of errors over time keeps increasing, or is beyond a reasonable threshold, an investigation might be required. The reasonable threshold might differ from cluster to cluster depending on what's happening in it, but 10% is a good starting point.
+
+As of v1.17.0, the Jaeger Operator will trigger a reconciliation loop only when the custom resource has changed or when the previous reconciliation has failed. Therefore, it's normal to have this metric with the same value over a long period of time: it only indicates that the custom resource hasn't changed. If this number keeps changing every second, it's indicative that something in the cluster is periodically changing the custom resource, or that the Jaeger Operator is undoing a change that is being done by a different component. Future versions of the Jaeger Operator might trigger a periodic reconciliation loop.
```
Comment (objectiser) [inline content/docs/1.17/operator.md]: "it's normal to have this metric with the same value..." - which metric is this referring to? Might be better to be explicit.

File: content/docs/1.17/operator.md (hunk 1 above)
Comment (jpkrohling) [reply content/docs/1.17/operator.md]: Absolutely! This is the default text that people see from `/metrics`, but this doc should really explain it in our context.

Comment (jpkrohling) [reply content/docs/1.17/operator.md]: How about this? 

> # How long (in seconds) the oldest reconciliation loop is taking. (gauge)

When there is only one reconciliation loop, this gauge shows how long the reconciliation is taking. When there are multiple, it shows the value for the oldest reconciliation loop (ie, the one that is taking the longest).

File: content/docs/1.17/operator.md (hunk 3 above)
Comment (jpkrohling) [reply content/docs/1.17/operator.md]: Changed the doc to be explicit about the other metrics as well.

**Comment by jpkrohling** [comment]
This is ready for another review.

**Pull request merged**